core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -119,12 +119,12 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
         // Compute the delay
         val clientMetric = metrics.metrics().get(clientRateMetricName(clientId))
         throttleTimeMs = throttleTime(clientMetric, getQuotaMetricConfig(quota(clientId)))
-        clientSensors.throttleTimeSensor.record(throttleTimeMs)
+        // If delayed, add the element to the delayQueue
         delayQueue.add(new ThrottledResponse(time, throttleTimeMs, callback))
         delayQueueSensor.record()
-        // If delayed, add the element to the delayQueue
         logger.debug("Quota violated for sensor (%s). Delay time: (%d)".format(clientSensors.quotaSensor.name(), throttleTimeMs))
     }
+    clientSensors.throttleTimeSensor.record(throttleTimeMs)
```

Comment:
Hmm, doesn't that skew the avg value? The admins probably only cares about the avg of the actual throttled times, not including the 0s. With the rest of the changes, it seems that you can just record throttleTimeSensor at where it was?